src/helpers/format.js

```diff
@@ -108,10 +127,122 @@ export function formatRecord(record, settings = []) {
+
+    debugData(`URN f856s: ${JSON.stringify(f856sUrn)}`);
+    debugData(`URN f856s with temp subfield: ${JSON.stringify(f856sUrnsWithTempSubfields)}`);
+
+    // None of the URNs has temp subfields, we don't need to do anything
+    if (f856sUrnsWithTempSubfields.length < 1) {
+      debug(`No Urns with temp subfield`);
+      return;
+    }
+
+    // Do we have an existing URN with legal deposit subfields, if we have, we can remove the temp URN field(s)
+    const existingLegalDepositURN = validateLD(f856sUrnsWithNoTempSubfields);
+    debug(`existingLegalDepositURN: ${existingLegalDepositURN}`);
+
+    if (existingLegalDepositURN) {
+      debug(`We have an existing LD URN, we can delete temp fields (${f856sUrnsWithTempSubfields.length})`);
+      newRecord.removeFields(f856sUrnsWithTempSubfields);
+      return;
+    }
+
+    // We do not have an existing legalDepositURN, we should use the tempURN and remove the temp subfields from it
+    if (!existingLegalDepositURN && f856sUrnsWithTempSubfields.length > 0) {
+      //debug(`All Urns (${f856sUrn.length}) have a temp subfield`);
+      debugData(`Original temp URNs: (${JSON.stringify(f856sUrnsWithTempSubfields)})`);
+      const fixedFields = f856sUrnsWithTempSubfields.map(removeTempSubfield);
+      debug(`We removed temp subfields: (${JSON.stringify(fixedFields)})`);
+      newRecord.removeFields(f856sUrnsWithTempSubfields);
+      newRecord.insertFields(fixedFields);
+      return;
+    }
+
+    // Should we check that the non-temp URN has 2nd ind '0' - meaning that the URN handles the actual resource itself?
+  }
+
+  // ---- LD-functions below are same as in https://github.com/NatLibFi/marc-record-validators-melinda/blob/feature-updates-to-urn/src/urn.js
+  // we propably should develop the urn/legaldeposit -validator to use also here
+
+  function createLDSubfields() {
+    return [
+      {
+        code: 'z',
+        value: 'Käytettävissä vapaakappalekirjastoissa'
+      },
+      {
+        code: '5',
+        value: 'FI-Vapaa'
+      }
+    ];
+  }
+
+  function validateLD(f856sUrn) {
+    debug(`Validating the existence of legal deposit subfields`);
+    const ldSubfields = createLDSubfields();
+    const f856sUrnWithLdSubfields = f856sUrn.filter(field => fieldHasLDSubfields(field, ldSubfields));
+    if (f856sUrnWithLdSubfields.length > 0) {
+      debug(`Record has ${f856sUrnWithLdSubfields.length} URN fields with all necessary legal deposit subfields`);
+      debugData(`f856sUrnWithLdSubfields: ${JSON.stringify(f856sUrnWithLdSubfields)}`);
+      return true;
+    }
+    return false;
+  }
+
+  function fieldHasLDSubfields(field, ldSubfields) {
+    if (ldSubfields.every(ldsf => field.subfields.some(sf => sf.code === ldsf.code && sf.value === ldsf.value))) {
+      return true;
+    }
+  }
+
+  // ---
+
+  function removeTempSubfield(field) {
+
+    // Sanity check for control fields
+    if (field.subfields === undefined) {
+      return field;
+    }
```

Comment:
If field has no subfields and is not control field.. it should be removed

Reply:
Fixed